tests/countries/test_brunei.py

```diff
@@ -0,0 +1,166 @@
+#  python-holidays
+#  ---------------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/dr-prodigy/python-holidays
+#  License: MIT (see LICENSE file)
+
+from holidays.countries.brunei import Brunei, BN, BRN
+from tests.common import TestCase
+
+
+class TestBrunei(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(
+            Brunei,
+            years=range(1984, 2077),
+            years_non_observed=range(1984, 2077),
+        )
+
+    def test_country_aliases(self):
+        self.assertCountryAliases(Brunei, BN, BRN)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Brunei(years=1983))
+
+    def test_special_holidays(self):
+        self.assertHoliday("2017-10-05")
+
+    def test_2022(self):
+        self.assertHolidays(
+            Brunei(years=2022),
+            ("2022-01-01", "Awal Tahun Masihi"),
+            ("2022-02-01", "Tahun Baru Cina"),
+            ("2022-02-23", "Hari Kebangsaan"),
+            ("2022-02-28", "Israk dan Mikraj"),
```

Comment:
On this, I don't think I can find any online site somewhere that actually shows Brunei's `exact_dates` for Islamic calendar.

That and it appears that the current `l10n` mechanism failed to work with Islamic Calendar's estimated dates properly - here are one of such example:
```python
            (
                ("2023-01-01", "New Year's Day"),
                ("2023-01-02", "New Year's Day (Observed)"),
                ("2023-01-22", "Lunar New Year"),
                ("2023-01-23", "Lunar New Year (Observed)"),
                ("2023-02-18", "Israk dan Mikraj* (*anggaran)"),
                ("2023-02-23", "National Day"),
                ("2023-03-23", "Hari Pertama Berpuasa* (*anggaran)"),
                ("2023-04-08", "Hari Nuzul Al-Quran* (*anggaran)"),
                ("2023-04-21", "Hari Raya Aidil Fitri* (*anggaran)"),
                ("2023-04-22", "Hari Raya Aidil Fitri* (*anggaran)"),
                ("2023-04-23", "Hari Raya Aidil Fitri* (*anggaran)"),
                ("2023-04-24", "Hari Raya Aidil Fitri (Observed)"),
                ("2023-05-31", "Armed Forces Day"),
                ("2023-06-28", "Hari Raya Aidil Adha* (*anggaran)"),
                ("2023-07-15", "Sultan Hassanal Bolkiah's Birthday"),
                ("2023-07-19", "Awal Tahun Hijrah* (*anggaran)"),
                ("2023-09-27", "Maulidur Rasul* (*anggaran)"),
                ("2023-12-25", "Christmas Day"),
            )
```
as the estimated code section only gives "`estimate string`" rather than "`%s* (*estimate string)`" @arkid15r

Reply:
Another issue to resolve in https://github.com/dr-prodigy/python-holidays/pull/1140?
I'll look into this soon.